blocks/library/heading/index.js

```diff
@@ -158,6 +163,25 @@ registerBlockType( 'core/heading', {
 						createBlock( 'core/paragraph', { content: after } ),
 					] );
 				} }
+				onReplace={ ( blocks ) => {
```

Comment:
Is this something that could be handled elsewhere? We cannot expect people to do these kind of operations in their blocks to have a consistent experience.

Reply:
I also think this should be part of the "pasting" handler because `onReplace` means we're replacing the current block, we can't expect it to be called while pasting only and thus we can't probably mess up the passed parameter.